src/svg.rs

```diff
@@ -373,4 +416,130 @@ mod tests {
+                Point::new(30., 30.),
+                Point::new(20., 20.),
+                Point::new(10., 10.),
+            )
+            .into(),
+        ];
+        let path = BezPath::from_path_segments(segments.iter().cloned());
+
+        assert_eq!(path.to_svg(), "M10 10C20 20 30 30 40 40C30 30 20 20 10 10");
+    }
+
+    #[test]
+    fn test_write_svg_two_move() {
+        let segments = [
+            CubicBez::new(
+                Point::new(10., 10.),
+                Point::new(20., 20.),
+                Point::new(30., 30.),
+                Point::new(40., 40.),
+            )
+            .into(),
+            CubicBez::new(
+                Point::new(50., 50.),
+                Point::new(30., 30.),
+                Point::new(20., 20.),
+                Point::new(10., 10.),
+            )
+            .into(),
+        ];
+        let path = BezPath::from_path_segments(segments.iter().cloned());
+
+        assert_eq!(
+            path.to_svg(),
+            "M10 10C20 20 30 30 40 40M50 50C30 30 20 20 10 10"
+        );
+    }
+
+    use rand::prelude::*;
+
+    fn gen_random_path_sequence(rng: &mut impl Rng) -> Vec<PathSeg> {
+        const MAX_LENGTH: u32 = 10;
+
+        let mut elements = vec![];
+        let mut position = None;
+
+        let length = rng.gen_range(0, MAX_LENGTH);
+        for _ in 0..length {
+            let should_follow: bool = rand::random();
+            let kind = rng.gen_range(0, 3);
+
+            let first = if should_follow && position.is_some() {
+                position.unwrap()
+            } else {
+                Point::new(rng.gen(), rng.gen())
+            };
+
+            let element: PathSeg = match kind {
+                0 => Line::new(first, Point::new(rng.gen(), rng.gen())).into(),
+
+                1 => QuadBez::new(
+                    first,
+                    Point::new(rng.gen(), rng.gen()),
+                    Point::new(rng.gen(), rng.gen()),
+                )
+                .into(),
+
+                2 => CubicBez::new(
+                    first,
+                    Point::new(rng.gen(), rng.gen()),
+                    Point::new(rng.gen(), rng.gen()),
+                    Point::new(rng.gen(), rng.gen()),
+                )
+                .into(),
+
+                _ => unreachable!(),
+            };
+
+            position = Some(element.end());
+            elements.push(element);
+        }
+
+        elements
+    }
+
+    #[test]
+    fn test_serialize_deserialize() {
+        const N_TESTS: u32 = 100;
+        let mut rng = rand::thread_rng();
+
+        for _ in 0..N_TESTS {
+            let vec = gen_random_path_sequence(&mut rng);
+            let ser = BezPath::from_path_segments(vec.iter().cloned()).to_svg();
+            let deser = BezPath::from_svg(&ser).expect("failed deserialization");
+
+            let deser_vec = deser.segments().collect::<Vec<PathSeg>>();
+
+            assert_eq!(vec, deser_vec);
```

Comment:
Generally it's not recommended to rely on floating point equality as there can be roundoff errors. I'm wondering though whether the guarantees of the underlying operations are strong enough though.

Otherwise we'd need to implement some kind of approximate equality, which is not conceptually difficult but is pretty tedious.